File: 이것이 취업을 위한 코딩 테스트다/그리디/모험가_길드/모험가_길드.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
  vector<int> fears;
  int n;
  cin >> n;

  for (int i = 0; i < n; i++) {
    int fear;
    cin >> fear;

    fears.push_back(fear);
  }

  sort(fears.begin(), fears.end());

  int result = 0;
  int count = 0;

  for (int i = 0; i < n; i++) {
    count += 1;

    if (count >= fears[i]) {
      result += 1;
      count = 0;
    }
  }

  cout << result;

  return 0;
}

File: 이것이 취업을 위한 코딩 테스트다/그리디/모험가_길드/모험가_길드.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class 모험가_길드 {
  public static ArrayList<Integer> fears = new ArrayList<>();
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();

    for (int i = 0; i < n; i++) {
      int fear = sc.nextInt();
      fears.add(fear);
    }

    Collections.sort(fears);

    int result = 0;
    int count = 0;

    for (int i = 0; i < n; i++) {
      count += 1;

      if (count >= fears.get(i)) {
        result += 1;
        count = 0;
      }
    }

    System.out.println(result);
  }
}
